google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -1730,6 +1988,168 @@ public TransactionManager transactionManager(TransactionOption... options) {
     }
   }
 
+  class MultiplexedSession implements CachedSession {
+    SessionImpl delegate;
+
+    MultiplexedSession(SessionImpl session) {
+      this.delegate = session;
+    }
+
+    @Override
+    public boolean isAllowReplacing() {
+      // for multiplexed session there is only 1 session, hence there is nothing that we
+      // can replace.
+      return false;
+    }
+
+    @Override
+    public void setAllowReplacing(boolean allowReplacing) {
+      // for multiplexed session there is only 1 session, there is nothing that can be replaced.
+      // hence this is no-op.
+    }
+
+    @Override
+    public void markBusy(ISpan span) {
+      this.delegate.setCurrentSpan(span);
```

Comment:
Isn't `delegate` the single `SessionImpl` instance that we have for our only multiplexed session? Meaning that calling this will overwrite the value that was set by the previous `MultiplexedSession` instance, even when that other `MultiplexedSession` is still being used for something?

Reply:
Yes, it will overwrite the value. But I also think the `setSpan` method is obsolete and won't have any impact on the traces. See this line in `ISpan span = tracer.spanBuilder(READ_ONLY_TRANSACTION);` for `singleUse` method.

So for each new singleUse() method call a new span is getting created. And I don't think calling this setter will impact how the traces would appear for concurrent transactions. I'll keep this comment open till I do a few more manual tests.

Reply:
If it is not needed, then we should remove it. This would otherwise just cause confusion.